### PR TITLE
perf(pbft): Remove duplicate PBFT synced blocks verification and getDagBlockOrder

### PR DESCRIPTION
**Pull request by gaoqi8649**
## Version Notes ##
{ major | minor | patch }

**NOTE: Include the version notes in the squash commit message**

### CODE REVIEW

**Review by justinsnapp — APPROVED** [review]

**Comment by JakubFornadel** [comment]
Not sure if this PR is supposed to optimize the stuff we were talking about on Thursday call, so the rest of the comment might not be relevant.

All this PR does is that now we dont compare parent hash of current synced pbft block twice, but only once. I mean yeah it does not need to be checked twice but it will have zero impact on performance. The real optimization we were talking about on Thursday call is that "getDagBlockOrder" is called twice, not this basic check...

File: src/consensus/pbft_manager.cpp [hunk 1]
```diff
@@ -1332,31 +1333,22 @@ void PbftManager::pushSyncedPbftBlocksIntoChain_() {
...
-      pbft_last_observed_synced_queue_size_ = pbft_synced_queue_size;
-      break;
-    }
+
     if (!comparePbftBlockScheduleWithDAGblocks_(*pbft_block_and_votes.pbft_blk)) {
+      LOG(log_nf_) << "DAG has not build up for anchor " << pbft_block_and_votes.pbft_blk->getPivotDagBlockHash()
+                   << " in PBFT block " << pbft_block_hash;
       break;
     }
     if (pushPbftBlock_(pbft_block_and_votes)) {
```
Comment (JakubFornadel) [inline src/consensus/pbft_manager.cpp:1348]: Here is the real perf problem. We call getDagBlockOrder inside "pushPbftBlock_" as well as in  "comparePbftBlockScheduleWithDAGblocks_", which is called few lines above...

**Review by JakubFornadel — CHANGES_REQUESTED** [review]
This commit does not fix anything, does it ? It could be perf as you optimized it so it does not compare that 1 pbft parent hash twice but still this wont affect performance in any manner so chore might be more appropriate... But I could live with perf too :) And dont use capital letters if possible - these commits will be automatically processed and changelog will be generated out of them. Thats why it is important to think about the proper type and write it in a form that was defined

**Comment by gaoqi8649** [comment]
> Not sure if this PR is supposed to optimize the stuff we were talking about on Thursday call, so the rest of the comment might not be relevant.
> 
> All this PR does is that now we dont compare parent hash of current synced pbft block twice, but only once. I mean yeah it does not need to be checked twice but it will have zero impact on performance. The real optimization we were talking about on Thursday call is that "getDagBlockOrder" is called twice, not this basic check...

Could you please show me where call the "getDagBlockOrder" twice? I didn't find that

File: src/consensus/pbft_manager.cpp (hunk 1 above)
Comment (gaoqi8649) [reply src/consensus/pbft_manager.cpp]: Thanks, fixed

**Comment by gaoqi8649** [comment]
> This commit does not fix anything, does it ? It could be perf as you optimized it so it does not compare that 1 pbft parent hash twice but still this wont affect performance in any manner so chore might be more appropriate... But I could live with perf too :) And dont use capital letters if possible - these commits will be automatically processed and changelog will be generated out of them. Thats why it is important to think about the proper type and write it in a form that was defined

Thanks! Fixed

**Review by JakubFornadel — APPROVED** [review]

**Pull request merged**